Tests/Data/RichardsMechanics/flow_fully_saturated_coordinate_system.prj

```diff
@@ -0,0 +1,291 @@
+                                    </coords>
+                                <values>4.43E-05 0.00026547 0.00159003 0.00325358 0.00665757 0.00952343 0.05704014 0.08159396 0.11671736 0.16696017 0.23883078 0.34163922 0.49931406 0.58449912 0.69907308 0.8361059 1
+                                    </values>
+                            </curve>
+                        </relative_permeability>
+                    </porous_medium>
+                </porous_medium>
+            </material_property>
+            <secondary_variables>
+                <secondary_variable type="static" internal_name="sigma" output_name="sigma"/>
+                <secondary_variable type="static" internal_name="epsilon" output_name="epsilon"/>
+                <secondary_variable type="static" internal_name="velocity" output_name="velocity"/>
+                <secondary_variable type="static" internal_name="saturation" output_name="saturation"/>
+            </secondary_variables>
+            <specific_body_force>0 0</specific_body_force>
+            <temperature> temp </temperature>
+        </process>
+    </processes>
+    <time_loop>
+        <processes>
+            <process ref="RM">
+                <nonlinear_solver>basic_newton</nonlinear_solver>
+                <convergence_criterion>
+                    <type>PerComponentDeltaX</type>
+                    <norm_type>NORM2</norm_type>
+                    <abstols>1e-14 1e-16 1e-16</abstols>
+                </convergence_criterion>
+                <time_discretization>
+                    <type>BackwardEuler</type>
+                </time_discretization>
+                <time_stepping>
+                    <type>FixedTimeStepping</type>
+                    <t_initial>0</t_initial>
+                    <t_end>2</t_end>
+                    <timesteps>
+                        <pair>
+                            <repeat>2</repeat>
+                            <delta_t>1</delta_t>
+                        </pair>
+                    </timesteps>
+                </time_stepping>
+            </process>
+        </processes>
+        <output>
+            <type>VTK</type>
+            <prefix>flow_fully_saturated_coordinate_system</prefix>
+            <timesteps>
+                <pair>
+                    <repeat>10000</repeat>
+                    <each_steps>1</each_steps>
+                </pair>
+            </timesteps>
+            <variables>
+                <variable>displacement</variable>
+                <variable>pressure</variable>
+                <variable>sigma</variable>
+                <variable>epsilon</variable>
+                <variable>velocity</variable>
+                <variable>saturation</variable>
+            </variables>
+        </output>
+    </time_loop>
+    <local_coordinate_system>
```

Comment:
There may be not only one local system in some cases. It might be good to use a nested elements as
```
<local_coordinate_system>
    <local_coordinate_system>
        <name> x1  </name>
        <basis_vector_0>e0</basis_vector_0>
        <basis_vector_1>e1</basis_vector_1>
    </local_coordinate_system>
</local_coordinate_system>
```

Reply:
@JobstM Material group dependent coordinate systems are not a problem with a single \<local_coordinate_system\> tag.This is because the parameters for the base vectors can be group based. An (untested) example:
```xml
<local_coordinate_system>
    <basis_vector_0>e0</basis_vector_0>
    <basis_vector_1>e1</basis_vector_1>
</local_coordinate_system>
<parameters>
    <parameter>
        <name>e0</name>
        <type>Group</type>
        <group_id_property>MaterialIDs</group_id_property>
        <index_values>
            <index>0</index>
            <values>1 0 0</value> <!-- e0 for first material -->
        </index_values>
        <index_values>
            <index>1</index>
            <value>0.707 0.707 0</value> <!-- e0 for second material -->
         </index_values>
    </parameter>
    <parameter>
        <name>e1</name>
        <type>Group</type>
        <group_id_property>MaterialIDs</group_id_property>
        <index_values>
            <index>0</index>
            <values>0 1 0</value> <!-- e1 for first material -->
        </index_values>
        <index_values>
            <index>1</index>
            <value>-0.707 0.707 0</value> <!-- e1 for second material -->
         </index_values>
    </parameter>
</parameters>
```

Reply:
More local_coordinate_system-tags are then only necesary if different parameters for the same material group have distinct principle directions? Or is there a way to represent this in the structure, too?

Reply:
I see. The current implementation can handle multi-local systems

@ThieJan Like this (figure from the Internet)
![untitled](https://user-images.githubusercontent.com/1343839/53243026-de168680-36a6-11e9-844a-69a644492a6c.png)

Reply:
That's correct, the multiple \<local_coordinate_system\> tags would only be necessary if, for example, temperature diffusion lives in a different coordinate system than the hydraulic permeability.

Suggest to wait until such a project comes up; the implementation is simple, just a name and instead of boolean \<use_local_coordinate_system\> a specific coordinate system would be referenced.

Reply:
Another possible input could be:
```xml
<local_coordinate_system>
    <basis_matrix>basis</basis_matrix>
</local_coordinate_system>
<parameters>
    <parameter>
        <name>basis</name>
        <type>Group</type>
        <group_id_property>MaterialIDs</group_id_property>
        <index_values>
            <index>0</index>
            <values>1 0 0 0 1 0 0 0 1</value> <!-- basis for first material -->
        </index_values>
        <index_values>
            <index>1</index>
            <values>0.707 0.707 0 -0.707 0.707 0 0 0 1</values> <!--  basis for second material -->
         </index_values>
    </parameter>
</parameters>
```